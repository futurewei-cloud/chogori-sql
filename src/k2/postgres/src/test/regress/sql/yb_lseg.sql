--
-- LSEG
-- Line segments
--

--DROP TABLE LSEG_TBL;
CREATE TABLE LSEG_TBL (id int primary key, s lseg);

INSERT INTO LSEG_TBL VALUES (1, '[(1,2),(3,4)]');
INSERT INTO LSEG_TBL VALUES (2, '(0,0),(6,6)');
INSERT INTO LSEG_TBL VALUES (3, '10,-10 ,-3,-4');
INSERT INTO LSEG_TBL VALUES (4, '[-1e6,2e2,3e5, -4e1]');
INSERT INTO LSEG_TBL VALUES (5, '(11,22,33,44)');

-- bad values for parser testing
INSERT INTO LSEG_TBL VALUES (6, '(3asdf,2 ,3,4r2)');
INSERT INTO LSEG_TBL VALUES (7, '[1,2,3, 4');
INSERT INTO LSEG_TBL VALUES (8, '[(,2),(3,4)]');
INSERT INTO LSEG_TBL VALUES (9, '[(1,2),(3,4)');

select * from LSEG_TBL ORDER BY id;

SELECT * FROM LSEG_TBL WHERE s <= lseg '[(1,2),(3,4)]' ORDER BY id;

SELECT * FROM LSEG_TBL WHERE (s <-> lseg '[(1,2),(3,4)]') < 10 ORDER BY id;
