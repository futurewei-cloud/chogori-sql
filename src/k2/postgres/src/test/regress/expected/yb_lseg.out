--
-- LSEG
-- Line segments
--
--DROP TABLE LSEG_TBL;
CREATE TABLE LSEG_TBL (id int primary key, s lseg);
INSERT INTO LSEG_TBL VALUES (1, '[(1,2),(3,4)]');
INSERT INTO LSEG_TBL VALUES (2, '(0,0),(6,6)');
INSERT INTO LSEG_TBL VALUES (3, '10,-10 ,-3,-4');
INSERT INTO LSEG_TBL VALUES (4, '[-1e6,2e2,3e5, -4e1]');
INSERT INTO LSEG_TBL VALUES (5, '(11,22,33,44)');
-- bad values for parser testing
INSERT INTO LSEG_TBL VALUES (6, '(3asdf,2 ,3,4r2)');
ERROR:  invalid input syntax for type lseg: "(3asdf,2 ,3,4r2)"
LINE 1: INSERT INTO LSEG_TBL VALUES (6, '(3asdf,2 ,3,4r2)');
                                        ^
INSERT INTO LSEG_TBL VALUES (7, '[1,2,3, 4');
ERROR:  invalid input syntax for type lseg: "[1,2,3, 4"
LINE 1: INSERT INTO LSEG_TBL VALUES (7, '[1,2,3, 4');
                                        ^
INSERT INTO LSEG_TBL VALUES (8, '[(,2),(3,4)]');
ERROR:  invalid input syntax for type lseg: "[(,2),(3,4)]"
LINE 1: INSERT INTO LSEG_TBL VALUES (8, '[(,2),(3,4)]');
                                        ^
INSERT INTO LSEG_TBL VALUES (9, '[(1,2),(3,4)');
ERROR:  invalid input syntax for type lseg: "[(1,2),(3,4)"
LINE 1: INSERT INTO LSEG_TBL VALUES (9, '[(1,2),(3,4)');
                                        ^
select * from LSEG_TBL ORDER BY id;
 id |               s
----+-------------------------------
  1 | [(1,2),(3,4)]
  2 | [(0,0),(6,6)]
  3 | [(10,-10),(-3,-4)]
  4 | [(-1000000,200),(300000,-40)]
  5 | [(11,22),(33,44)]
(5 rows)

SELECT * FROM LSEG_TBL WHERE s <= lseg '[(1,2),(3,4)]' ORDER BY id;
 id |       s
----+---------------
  1 | [(1,2),(3,4)]
(1 row)

SELECT * FROM LSEG_TBL WHERE (s <-> lseg '[(1,2),(3,4)]') < 10 ORDER BY id;
 id |         s
----+--------------------
  1 | [(1,2),(3,4)]
  2 | [(0,0),(6,6)]
  3 | [(10,-10),(-3,-4)]
(3 rows)
